spec/importc.dd

```diff
@@ -4,7 +4,164 @@ $(SPEC_S ImportC,
 
 $(HEADERNAV_TOC)
 
-$(P Documentation for ImportC coming soon!)
+    $(P ImportC is a C compiler embedded into the D implementation.
+    It's purpose is to enable direct importation of C files, without
+    needing to manually prepare a D file corresponding to the the declarations
+    in the C file. It enables directly compiling C files into modules, and
+    linking them in with D code to form an executable. It can be used
+    as a C compiler to compile and link 100% C programs.
+    )
+
+    $(P This is not a C reference manual nor programming tutorial.
+    It describes the specifics of the dialect of C that
+    ImportC is, and how to use it effectively.)
+
+
+$(H2 $(LNAME2 dialect, ImportC Dialect))
+
+    $(P There are many versions of C. ImportC is designed to be an
+    implementation of (LINK2 https://en.wikipedia.org/wiki/C11_(C_standard_revision), ISO/IEC 9899:1999),
+    which will hereafter be referred to as $(B C11).
+    References to the C11 Standard will be C11 followed by the paragraph number.
+    Prior versions, such as C99, C89, and K+R C, are not supported.
+    )
+
+    $(P Adjustment to the ImportC dialect is made to match the
+    behavior of the C compiler that the D compiler is matched to. For
+    example, on Win32 D is matched to the Digital Mars C compiler,
+    and can be matched to the Visual C compiler using the $(TT -mscoff32)
+    switch. Win64 D is matched to the Visual C compiler.
+    On Posix targets, the matching C compiler is Gnu C or Clang C.
+    )
+
+    $(P Further adjustment is made to take advantage of some of the D
+    implementation's capabilities.)
+
+    $(P This is all covered in the rest of this document.)
+
+
+$(H2 $(LNAME2 command-line, Invoking ImportC))
+
+    $(P The ImportC compiler can be invoked in two ways:)
+
+    $(UL
+    $(LI directly via the command line)
+    $(LI indirectly via importing a C file)
+    )
+
+    $(H3 ImportC Files on the Command Line)
+
+    $(H3 Importing C Files)
+
+
+$(H2 $(LNAME2 preprocessor, Preprocessor))
+
+    $(P ImportC does not have a preprocessor. It is designed to compile C
+    files after they have been first run through the C preprocessor.)
```

Comment:
Is this temporary? I thought that being able to include system headers was a big reason for this entire endeavour.

Reply:
Atila and I have plans in that direction, but they're not in the near future.

Reply:
> Is this temporary? I thought that being able to include system headers was a big reason for this entire endeavour.

Granted that we don't know all imports until after semantic3, how will that possibly scale? Other than being N times slower than a C compiler for needing to invoke CPP on every import seen.

Not to forget that system headers are littered with dialect-specific alternative keywords (`const`, `__const`, `__const__`) and ISO C extensions (ISO 18661-3:2015), so there's got to be a middle-man in-between the header and importC to [sanitize things](https://github.com/ibuclaw/importC/blob/main/src/keywords.c.in).

Reply:
> Other than being N times slower than a C compiler for needing to invoke CPP on every import seen.

It's not happening once per .h file. It's happening once per .h file *along with its #includes*, so it is exactly like what a C compiler would do.

> so there's got to be a middle-man in-between the header and importC

I added a section on wrapping files in a later PR, which addresses that issue.

Reply:
> It's not happening once per .h file. It's happening once per .h file along with its #includes, so it is exactly like what a C compiler would do.

A C compiler only calls `cpp` _once_, not once per thing.

What I'm refering to is:
```
import header1; // header1.h preprocessed and imported before semantic1
int fn()
{
    import header2; // header2.h preprocessed and imported during semantic3
}
```
That is to say, things won't scale well if we make the D compiler call the preprocessor too.

Reply:
> A C compiler only calls cpp once, not once per thing.

Yes, I know how C compilers work :-)

> What I'm refering to is:

What I'm saying is that for:
```
import header1;
int fn()
{
    import header1;
}
```
then `header1` is only processed **once**. Just as in D, `header1` can be imported again and again, yet it is only processed once. (Unlike in C.) Now, if the user has *n* `.c` files, and each `#include`'s `stdio.h`, then `studio.h` will be preprocessed *n* times, just as in C.

There is no reliable way to treat `#include` files as modules (I've tried, and so have others), and ImportC is not an attempt to fix any of C's problems (we have DasBetterC for that).

Reply:
In other words, I do not know of any case where an ImportC file will be preprocessed by D more often than a C compiler would, and many cases when it would be less.